bundles/org.openhab.core.io.rest.update/src/main/java/org/openhab/core/io/rest/update/internal/updaters/BaseUpdater.java

```diff
@@ -0,0 +1,810 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.io.rest.update.internal.updaters;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.math.BigInteger;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamConstants;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamReader;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.core.OpenHAB;
+import org.openhab.core.io.rest.update.internal.dto.OpenHabVersionDTO;
+import org.openhab.core.io.rest.update.internal.dto.UpdaterStatusDTO;
+import org.openhab.core.io.rest.update.internal.enums.TriState;
+import org.openhab.core.io.rest.update.internal.enums.VersionType;
+import org.osgi.framework.FrameworkUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.jcraft.jsch.ChannelExec;
+import com.jcraft.jsch.JSch;
+import com.jcraft.jsch.JSchException;
+import com.jcraft.jsch.Session;
+
+/**
+ * The {@link BaseUpdater} is a base for extending classes that will update openHAB.
+ *
+ * For each different operating system (resp. Linux package manager variety) there will be a different class that
+ * extends this class and executes the respective update process for the target system.
+ *
+ * @author AndrewFG - Initial contribution
+ */
+@NonNullByDefault
+public abstract class BaseUpdater implements Runnable {
+
+    protected final Logger logger = LoggerFactory.getLogger(BaseUpdater.class);
```

Comment:
Loggers should be private and you should have one per subclass as otherwise the logs will point to the wrong class.

Reply:
Ok. Done.